bazel/rules/cc/defs.bzl

```diff
@@ -6,7 +6,11 @@ def _copts(std):
             "/std:" + std,
             "/Zc:preprocessor",
         ],
-        "//conditions:default": ["-std=" + std],
+        "//conditions:default": [
+            "-std=" + std,
+            "-Werror",
+            "-fno-strict-aliasing",
```

Comment:
Where are we running into type punning on the host side? I'd prefer to avoid this on the host-side.